core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java

```diff
@@ -132,13 +136,30 @@ public Sep31PostTransactionResponse postTransaction(
 
     return Sep31PostTransactionResponse.builder()
         .id(txn.getId())
-        .stellarAccountId(asset.getDistributionAccount())
+        .stellarAccountId(Context.get().getAsset().getDistributionAccount())
         .stellarMemo(txn.getStellarMemo())
         .stellarMemoType(txn.getStellarMemoType())
         .build();
   }
 
-  private void generateTransactionMemo(Sep31Transaction txn) throws SepException {
+  private void applyAmount() {
```

Comment:
💡 maybe we should rename it to `calculateAmountIn()`? There are 3 types of amount for in, out and fee, so amount is not specific enough for this method name.

Reply:
I can refactor the calculation if the diagram is accurate. https://lucid.app/lucidchart/55d25dfd-e895-4347-bea3-15a05bb6ccfa/edit?page=ZXw_~8n.uJ7c&invitationId=inv_7ae1fca7-04d1-4023-a650-8606c5627ed8#